solc/CommandLineParser.cpp

```diff
@@ -887,7 +898,8 @@ void CommandLineParser::processArgs()
 	if (
 		m_options.input.mode == InputMode::Help ||
 		m_options.input.mode == InputMode::License ||
-		m_options.input.mode == InputMode::Version
+		m_options.input.mode == InputMode::Version ||
+		m_options.input.mode == InputMode::LanguageServer
```

Comment:
This returns early now, but it currently does not check for invalid options. Can you quickly tell me how to do this easily, @cameel ?

Reply:
I think you should return a bit father down below. Just below this part there is a stub of a check against options disallowed in specific modes:

https://github.com/ethereum/solidity/blob/248bc387cdf39eb314536cc959000f122d32b179/solc/CommandLineParser.cpp#L894-L904

If `--lsp` does not have any options, you can exit immediately after that part. If it does support some, you should list them in the map and insert the return later, only after all those options have been processed. This might require some reordering if the code processing them is mixed with code processing options LSP does not use.

Reply:
Ok, I can move it down. There are zero options, the lsp is configured through json-rpc messages.

Reply:
Looks fine now. This should start rejecting all non-LSP options once we actually list them here but that will be covered by #12166 eventually.